packages/react-core/src/components/Select/SelectOption.tsx

```diff
@@ -228,166 +231,209 @@ export class SelectOption extends React.Component<SelectOptionProps> {
-                )}
-                ref={this.liRef}
-              >
-                {isLoading && children}
-                {!isLoading && (
-                  <>
-                    <Component
-                      {...props}
-                      className={css(
-                        styles.selectMenuItem,
-                        isLoad && styles.modifiers.load,
-                        isSelected && styles.modifiers.selected,
-                        isDisabled && styles.modifiers.disabled,
-                        description && styles.modifiers.description,
-                        isFavorite !== null && styles.modifiers.link,
-                        className
-                      )}
-                      onClick={(event: any) => {
-                        if (isLoad) {
-                          onViewMoreClick(event);
-                          event.stopPropagation();
-                        } else if (!isDisabled && !isLoading) {
-                          onClick(event);
-                          onSelect(event, value, isPlaceholder);
-                          onClose();
-                        }
-                      }}
-                      role="option"
-                      aria-selected={isSelected || null}
-                      ref={this.ref}
-                      onKeyDown={(event: React.KeyboardEvent) => {
-                        this.onKeyDown(event, 0);
-                      }}
-                      type="button"
-                    >
-                      {description && (
-                        <React.Fragment>
-                          <span className={css(styles.selectMenuItemMain)}>
-                            {itemDisplay}
-                            {isSelected && (
-                              <span className={css(styles.selectMenuItemIcon)}>
-                                <CheckIcon aria-hidden />
-                              </span>
-                            )}
-                          </span>
-                          <span className={css(styles.selectMenuItemDescription)}>{description}</span>
-                        </React.Fragment>
-                      )}
-                      {!description && (
-                        <React.Fragment>
-                          {itemDisplay}
-                          {isSelected && (
-                            <span className={css(styles.selectMenuItemIcon)}>
-                              <CheckIcon aria-hidden />
-                            </span>
-                          )}
-                        </React.Fragment>
-                      )}
-                    </Component>
-                    {isFavorite !== null && id && favoriteButton(onFavorite)}
-                  </>
-                )}
-              </li>
+    const renderOption = (onSelect: any, onClose: any, variant: string, inputIdPrefix: string, onFavorite: any) => {
+      if (variant !== SelectVariant.checkbox && isLoading && isGrouped) {
```

Comment:
Could you add a one line comment to describe the situation for each of these conditionals?

Reply:
I don't know that that is necessary.  I just copied what was there before and move it into a function.  the comment would also pretty much just state what is in the if statement.

Reply:
okay - I'm just imagining future scenarios where we want to make a small change, and trying to figure out what might be the side effects or what other logic scenarios exist would be made a lot easier if they were described in plain english a bit?